### PR TITLE
MGMT-5266 Add API endpoint for resetting failed host validation

### PR DESCRIPTION
**Pull request by ori-amizur**
The reset operation is for validations that failed in preparing for installation
/cc @avishayt 
/cc @jtomasek

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1888,6 +1888,57 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /clusters/{cluster_id}/hosts/{host_id}/actions/reset-validation/{validation_id}:
+    patch:
+      tags:
+        - installer
+      summary: Reset failed validation.
+      operationId: ResetValidation
```
Comment (avishayt) [inline swagger.yaml:1896]:
```suggestion
      operationId: ResetHostValidation
```

File: swagger.yaml [hunk 2]
```diff
@@ -1888,6 +1888,57 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /clusters/{cluster_id}/hosts/{host_id}/actions/reset-validation/{validation_id}:
+    patch:
+      tags:
+        - installer
+      summary: Reset failed validation.
```
Comment (avishayt) [inline swagger.yaml:1895]:
```suggestion
      summary: Reset failed host validation.
```

File: swagger.yaml [hunk 3]
```diff
@@ -1888,6 +1888,57 @@ paths:
...
+      operationId: ResetValidation
+      description: Reset failed validation.  The validation is executed during "preparing for installation" stage.
```
Comment (avishayt) [inline swagger.yaml:1897]:
```suggestion
      description: Reset failed host validation.  The validation is executed during "preparing for installation" stage.
```

Will it only work for validations executed during "preparing for installation" stage?

File: swagger.yaml (hunk 2 above)
Comment (ori-amizur) [reply swagger.yaml:1895]: OK

File: swagger.yaml (hunk 3 above)
Comment (ori-amizur) [reply swagger.yaml:1897]: That's the plan

File: swagger.yaml (hunk 1 above)
Comment (ori-amizur) [reply swagger.yaml:1896]: OK

File: swagger.yaml (hunk 3 above)
Comment (ori-amizur) [reply swagger.yaml:1897]: That's the plan

Comment (avishayt) [reply swagger.yaml:1897]: Why limit it?

Comment (ori-amizur) [reply swagger.yaml]: I am not sure it makes sense for the others.  These validations run only once.  Others will run periodically anyway.

Comment (ori-amizur) [reply swagger.yaml]: Although I can remove the mention of "preparing for installation".

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

**Review by avishayt — APPROVED** [review]

**Pull request merged**